src/http/server.ts

```diff
@@ -758,32 +758,35 @@ export class Server {
   protected getResourceClass(
     request: Drash.Http.Request,
   ): Drash.Interfaces.Resource | undefined {
-    for (const resourceName in this.resources) {
-      const resource = this.resources[resourceName];
-      const pathObjs = resource.paths_parsed!;
-      for (const pathObj of pathObjs) {
-        if (pathObj.og_path === "/" && request.url_path === "/") {
-          return resource;
-        }
-
-        const pathMatchesRequestPathname = request.url_path.match(
-          pathObj.regex_path,
-        );
-        if (pathMatchesRequestPathname == null) {
-          continue;
-        }
-
-        pathMatchesRequestPathname.shift();
-        const pathParamsInKvpForm: { [key: string]: string } = {};
-        pathObj.params.forEach((paramName: string, index: number) => {
-          pathParamsInKvpForm[paramName] = pathMatchesRequestPathname[index];
-        });
+    let resource = undefined;
```

Comment:
We should avoid using `let`, if possible, because it leads to a possible variable assignment deep in the logic that we might forget about. This could be avoided with early return statements.
If possible, I would prefer if a quick refactor in this function.

Reply:
this makes sense. do we want to move toward a functional approach in general? 

@crookse can you link me to the test file for this function?

Reply:
@Guergeiro, 00c7219

@saragee3, we could go with a functional approach, but this is the only place this kind of iteration happens? here's the test file and the test suite that covers this method: https://github.com/drashland/deno-drash/blob/7be08137fa581c7089f604117aa06ee48d1af585/tests/unit/http/server_test.ts#L55